### ISSUE TITLE
ShopGuiPlus error

### ISSUE THREAD

**Issue by Distolfix** [BG-Software-LLC/SuperiorSkyblock2#1645]
### Minecraft's Version

Purpur 1.19.3

### Plugin's Version

#982 24/03/2023 09:09:14

### Describe the bug

The skyblock plugin doesn't load shopguiplus first. This results in a buggy is top, for someone like me who set the prices from the shopgui plugin.

### To Reproduce

[11:02:55 INFO]: [EnchantGui] Total loaded enchantments: 40
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER:ZOMBIE
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER:ZOMBIE
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER:ZOMBIE
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER:COW
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER:COW
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER:SKELETON
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER:SKELETON
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER:BLAZE
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SPAWNER:BLAZE
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item DIORITE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item DIORITE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item ANDESITE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item ANDESITE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item STONE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item STONE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item RED_TULIP:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item RED_TULIP:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item COBBLESTONE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item COBBLESTONE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item DIRT:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item DIRT:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item ORANGE_TULIP:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item ORANGE_TULIP:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item WHITE_WOOL:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item WHITE_WOOL:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item GRANITE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item GRANITE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item GLASS:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item GLASS:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item GRASS_BLOCK:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item GRASS_BLOCK:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item PINK_TULIP:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item PINK_TULIP:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SAND:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SAND:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item CACTUS:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item CACTUS:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item OAK_LEAVES:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item OAK_LEAVES:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item PODZOL:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item PODZOL:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item COAL_BLOCK:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item COAL_BLOCK:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item STRIPPED_OAK_LOG:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item STRIPPED_OAK_LOG:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item DEEPSLATE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item DEEPSLATE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item AZURE_BLUET:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item AZURE_BLUET:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item WHITE_TULIP:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item WHITE_TULIP:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SMOOTH_STONE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SMOOTH_STONE:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item DANDELION:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item DANDELION:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item BEDROCK
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SUNFLOWER:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item SUNFLOWER:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item OAK_LOG:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item OAK_LOG:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item OXEYE_DAISY:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item OXEYE_DAISY:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item POPPY:0
[11:02:55 WARN]: [SuperiorSkyblock2] net.brcdev.shopgui.exception.shop.ShopsNotLoadedException: Shops aren't loaded yetFailed to load prices for item POPPY:0
[11:02:55 INFO]:  WhiteOmbrah_ è entrato
[11:02:55 INFO]: WhiteOmbrah_[/151.27.221.89:47192] logged in with entity id 184 at ([SuperiorWorld]0.9572659657576457, 100.0, 3596.447163330497)
[11:02:55 INFO]: [CMI] CMIB proxy plugin detected
[11:02:55 INFO]: [AuthMe] WhiteOmbrah_ logged in 151.27.221.89
[11:02:56 INFO]: [ShopGUIPlus] ================================[ ShopGUI+ 1.82.1 ]================================
[11:02:56 INFO]: [ShopGUIPlus]  
[11:02:56 INFO]: [ShopGUIPlus] Vault economy registered.
[11:02:56 INFO]: [ShopGUIPlus] Vault economy enabled.
[11:02:56 INFO]: [ShopGUIPlus] Using Vault as default economy provider.
[11:02:56 INFO]: [ShopGUIPlus] Permissions support enabled.
[11:02:56 INFO]: [ShopGUIPlus] Enabled item provider for ItemsAdder.
[11:02:56 INFO]: [ShopGUIPlus] Using WildStacker for spawners support.
[11:02:56 INFO]: [ShopGUIPlus] Loaded 6 main menu items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded shop 'miscellaneous' with 37 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded shop 'farming' with 69 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded shop 'ores' with 22 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded shop 'blocks' with 100 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded shop 'drops' with 9 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded shop 'tools' with 15 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded shop 'spawner' with 13 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded shop 'dyes' with 16 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded shop 'food' with 19 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded 9 shops with total of 300 items.
[11:02:57 INFO]: [ShopGUIPlus] Loaded 0 permission-based price modifiers.
[11:02:57 INFO]: [ShopGUIPlus]  
[11:02:57 INFO]: [ShopGUIPlus] ==============================================================================

### Additional Information

As you can see from the logs the plugin loads after the prices so the is top will be wrong until I enter and do /is recalc

**Comment by OmerBenGera** [comment]
SuperiorSkyblock waits for all the shops to be loaded before it interacts with SGP.
However, if the shops are not loaded until the first player that joins, it will forcefully try to interact with the shops, which might cause the error.
Therefore, the scenario you mentioned should not occur at all, as in both ways SGP will be loaded.
Make sure you update SGP to their latest version and see if it still occurs.

**Comment by Distolfix** [comment]
Bro ShopGuiplus is on the latest version.  ShopGUI+ 1.82.1 

![image](https://user-images.githubusercontent.com/73421268/227495843-dcd448b8-e3c6-414a-b229-4b020646ad89.png)

**Comment by OmerBenGera** [comment]
> Bro ShopGuiplus is on the latest version. ShopGUI+ 1.82.1
> 
> ![image](https://user-images.githubusercontent.com/73421268/227495843-dcd448b8-e3c6-414a-b229-4b020646ad89.png)

Try it on a server with default config files of SSB and SGP, let me know if you can replicate the issue

**Comment by Distolfix** [comment]
I tried with the default config of SuperiorSkyblock and i have the same problem

**Comment by OmerBenGera** [comment]
> I tried with the default config of SuperiorSkyblock and i have the same problem

Try it on a server with only SSB and SGP, both have default config files.

**Comment by Distolfix** [comment]
Sorry for my bad english bro

**Comment by OmerBenGera** [comment]
Check out latest dev build, let me know if it works

**Comment by Distolfix** [comment]
Waits bro

**Comment by Distolfix** [comment]
https://pastebin.com/g8RRx2Dz

**Comment by OmerBenGera** [comment]
> https://pastebin.com/g8RRx2Dz

SGP's event was not called and the prices were fetched as soon as you joined the server.
This is not a fault of SSB but SGP's

**Comment by Distolfix** [comment]
it's the exact same problem as the video I sent you it doesn't load the prices correctly

**Comment by OmerBenGera** [comment]
> it's the exact same problem as the video I sent you it doesn't load the prices correctly

Send the entire logs file please.
Upload it to pastebin and send it here

**Comment by Distolfix** [comment]
https://pastebin.com/0mHbV1yj

**Comment by OmerBenGera** [comment]
Shops are loaded only after you log in for no reason.
Check why it occurs with ShopGUIPlus's developer and report back.

**Comment by Distolfix** [comment]
it's not my log that causes the prices to be loaded, it's just that I enter as soon as the server opens today I'll show you

**Comment by OmerBenGera** [comment]
> it's not my log that causes the prices to be loaded, it's just that I enter as soon as the server opens today I'll show you

Then maybe wait a few moments, let the shops data load.

**Comment by Distolfix** [comment]
There is another bug I assure you that now I will show you in a video...

**Comment by OmerBenGera** [comment]
> There is another bug I assure you that now I will show you in a video...

Open a new issue about the other bug, do not mix different bugs in the same report.
Does it work if you wait a few moments before you join the server?

**Comment by Distolfix** [comment]
Yes, works but it's strange if one enters immediately it bugs how can I fix it

**Comment by OmerBenGera** [comment]
Contact ShopGUIPlus's developer regarding this, ask them why it takes so long for the shops data to be loaded.
They should load it before players join the server.

**Comment by Distolfix** [comment]
Can i link to Shopguiplus developer this link of github?

**Comment by Distolfix** [comment]
this discussion

**Comment by Distolfix** [comment]
Because it seems strange to me the problem is not if the player enters after the server writes Done. The problem is that if a player tries to enter while the server is loading as soon as Done, it will still give an error. Even if the player Enters while the server is loading and exits while loading

**Comment by OmerBenGera** [comment]
> Can i link to Shopguiplus developer this link of github?

Of course.

> Because it seems strange to me the problem is not if the player enters after the server writes Done. The problem is that if a player tries to enter while the server is loading as soon as Done, it will still give an error. Even if the player Enters while the server is loading and exits while loading

Right, as SSB will load prices as soon as SGP loads its data, or when the first player joins.
I am not sure why it takes so long for SGP to load their data that players manage to join before its done.

**Comment by Distolfix** [comment]
Also look at this problem. Which I'm linking you now. The spawners (is levels) are not loaded correctly by your plugin I don't know why but the zombie one works correctly at least for me.

The blocks as you can see are loading correctly. But spawners don't. The is level doesn't go up while the value does!

**Comment by Distolfix** [comment]
I send you a video one moment

**Comment by Distolfix** [comment]
I also tried with default test server (the server of yesterday) and i have the same problem. 

https://user-images.githubusercontent.com/73421268/227718366-86aae316-8811-479b-ba64-69957b280509.mp4

**Comment by Distolfix** [comment]
As you can see Zombie spawner loading correctly the is value and is levels. But the other spawner no

**Comment by OmerBenGera** [comment]
Only worth values are synced with ShopGUIPlus. Levels are taken from your levels.yml file or by using the formula in the config file. Configure your server correctly so other spawners will have level values as well.
This is not a bug.

Anyways, not related to the original issue, so keep this issue related only to the original bug report.

**Comment by Distolfix** [comment]
I have an idea to solve the bug I have. Instead of loading your plugin after the shop, as it causes issues, wouldn't it be better to establish the connection after the server has done loading? This way, ShopGUIPlus will still work, and the shops will load after the server says "Done." If you do the same, this bug might be resolved without the need for third-party intervention.

**Comment by OmerBenGera** [comment]
> I have an idea to solve the bug I have. Instead of loading your plugin after the shop, as it causes issues, wouldn't it be better to establish the connection after the server has done loading? This way, ShopGUIPlus will still work, and the shops will load after the server says "Done." If you do the same, this bug might be resolved without the need for third-party intervention. It's ridiculous that if a player tries to join before the server is open, they will spam the error message. Then, I have to manually perform an admin reload, which is unmanageable.

Contact SGP regarding this.
They should load data before players join the game.

**Comment by Distolfix** [comment]
Dear developer at this link https://github.com/brcdev-minecraft/shopgui-api there is a method to Hooking into ShopGUI+ by developer. 

This guarantees your code gets executed when ShopGUI+ had started up, but not loaded its shops yet, so you can register your items, spawners, economies etc. safely.

(Message from Shopguiplus wiki)

**Comment by brcdev** [comment]
> > I have an idea to solve the bug I have. Instead of loading your plugin after the shop, as it causes issues, wouldn't it be better to establish the connection after the server has done loading? This way, ShopGUIPlus will still work, and the shops will load after the server says "Done." If you do the same, this bug might be resolved without the need for third-party intervention. It's ridiculous that if a player tries to join before the server is open, they will spam the error message. Then, I have to manually perform an admin reload, which is unmanageable.
> 
> Contact SGP regarding this. They should load data before players join the game.

Unfortunately we can't as some plugins ShopGUI+ depends on don't do this either. I've provided a more exact description in the ticket linked above, but tl;dr this way:

> Dear developer at this link https://github.com/brcdev-minecraft/shopgui-api there is a method to Hooking into ShopGUI+ by developer.
> 
> This guarantees your code gets executed when ShopGUI+ had started up, but not loaded its shops yet, so you can register your items, spawners, economies etc. safely.
> 
> (Message from Shopguiplus wiki)

is a correct way to hook into SG+ and the only one that we can support/improve. If any further help is needed feel free to either contact me via any of the tickets or directly at Discord. Thanks!

**Comment by Distolfix** [comment]
It’s ok Omer. I think you can improve the bug with the metoth of Shopguiplus wiki! I hope I have helped you. and that you can fix this problem

**Comment by OmerBenGera** [comment]
> > > I have an idea to solve the bug I have. Instead of loading your plugin after the shop, as it causes issues, wouldn't it be better to establish the connection after the server has done loading? This way, ShopGUIPlus will still work, and the shops will load after the server says "Done." If you do the same, this bug might be resolved without the need for third-party intervention. It's ridiculous that if a player tries to join before the server is open, they will spam the error message. Then, I have to manually perform an admin reload, which is unmanageable.
> > 
> > 
> > Contact SGP regarding this. They should load data before players join the game.
> 
> Unfortunately we can't as some plugins ShopGUI+ depends on don't do this either. I've provided a more exact description in the ticket linked above, but tl;dr this way:
> 
> > Dear developer at this link https://github.com/brcdev-minecraft/shopgui-api there is a method to Hooking into ShopGUI+ by developer.
> > This guarantees your code gets executed when ShopGUI+ had started up, but not loaded its shops yet, so you can register your items, spawners, economies etc. safely.
> > (Message from Shopguiplus wiki)
> 
> is a correct way to hook into SG+ and the only one that we can support/improve. If any further help is needed feel free to either contact me via any of the tickets or directly at Discord. Thanks!

I am already listening to ShopsPostLoadEvent.
The issue is that you load the data after players join, which is weird and causes issues, as SSB will forcely try to access shops if player joins and the data wasnt loaded.

It used to work before, you changed something that broke this.

**Comment by OmerBenGera** [comment]
For rerefernce:
https://github.com/BG-Software-LLC/SuperiorSkyblock2/blob/master/Hooks/ShopGUIPlus78/src/main/java/com/bgsoftware/superiorskyblock/external/prices/PricesProvider_ShopGUIPlus78.java

**Comment by BlackGiok** [comment]
I have the same issue in paper spigot 1.18

**Comment by Distolfix** [comment]
blackgiok, we await news from Omar, hoping that he will be able to resolve this problem as soon as possible. from what I understand it's a shopguiplus bug, I hope they get up and running soon

**Comment by OmerBenGera** [comment]
@Distolfix @BlackGiok Try it on latest dev build.
Let me know if its better now.

**Comment by Distolfix** [comment]
Not work

**Comment by Distolfix** [comment]
The console spam error failed to load price but the prices after join are load correctly

**Comment by Distolfix** [comment]
[latest.log](https://github.com/BG-Software-LLC/SuperiorSkyblock2/files/11160853/latest.log)

**Comment by Distolfix** [comment]
[latest.log](https://github.com/BG-Software-LLC/SuperiorSkyblock2/files/11160856/latest.log)

**Comment by OmerBenGera** [comment]
@brcdev When are you loading the shops?
I changed it to load the shops after a second the server starts and they are still not loaded...

**Comment by Distolfix** [comment]
@brcdev @OmerBenGera I have also updated ShopGuiPlus to the latest version, but the problem still persists. It's the same error as before. I'm looking forward to any updates you may have.

**Comment by Distolfix** [comment]
There is a problem, sometimes after a restart I don't know what caused it, the plugin goes into a loop at startup spamming warnings and not allowing entry. I can't even kill the server and I'm forced to restart the machine. The problem seems more serious than expected. We need to come to an agreement or we may have to remove the prices from shopguiplus.

**Comment by OmerBenGera** [comment]
@Distolfix Check out latest dev build, let me know if it's better.
If not, then send me the logs again.

**Comment by Distolfix** [comment]
Works Thanks!

**Comment by Distolfix** [comment]
[latest.log](https://github.com/BG-Software-LLC/SuperiorSkyblock2/files/11209526/latest.log)


bug returned

**Comment by Distolfix** [comment]
@OmerBenGera

**Comment by OmerBenGera** [comment]
@brcdev The data is loaded like 5 seconds after the server has started...
That's obviously an issue on your end.

**Comment by Distolfix** [comment]
https://pastebin.com/rRYfu2gd

And server go offline

**Comment by Distolfix** [comment]
And then crash the server

https://pastebin.com/rRYfu2gd

**Comment by Distolfix** [comment]
Sorry but WTF 

![image](https://user-images.githubusercontent.com/73421268/231518417-519a2979-d236-41d6-be57-60929dc2e5d2.png)

2022?

**Comment by Distolfix** [comment]
![image](https://user-images.githubusercontent.com/73421268/231519111-667db553-9979-4c31-81dc-b3c260540645.png)

Bro why the dev is 2022?

**Comment by Distolfix** [comment]
I close the ticket. If the problem comes back I will reopen it

**Comment by Distolfix** [comment]
https://pastebin.com/0aHdT3Ur

**Comment by Distolfix** [comment]
https://pastebin.com/0aHdT3Ur

**Comment by Distolfix** [comment]
@OmerBenGera

**Comment by OmerBenGera** [comment]
Not related to the original issue in any way.
Open a new issue, provide there your SGP version and everything else.
